native/src/main/java/io/ballerina/lib/avro/Utils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.avro;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BError;
+
+import static io.ballerina.lib.avro.ModuleUtils.getModule;
+
+public class Utils {
+
+    private Utils() {
+    }
+
+    public static final String ERROR_TYPE = "Error";
+    public static final String AVRO_SCHEMA = "avroSchema";
+
+    public static final String JSON_PROCESSING_ERROR = "JSON processing error";
+    public static final String DESERIALIZATION_ERROR = "Unable to deserialize the byte value";
+
+
+    public static BError createError(String message, Throwable throwable) {
+        BError cause = throwable == null ? null : ErrorCreator.createError(throwable);
```

Comment:
Can't we have another method like following rather than this:
```java
public static BError createError(String message) {
  // logic
}
```

Reply:
Instead of adding a new method here, the nullable check is removed from this method as the exception instance is always non null.